lib/features/coin_flow/screens/search_for_coin_screen.dart

```diff
@@ -1,49 +1,129 @@
-                        eventName: AmplitudeEvent.buttonPressed,
-                        eventProperties: {
-                          'button_name': 'Assign the coin',
-                          'formatted_date': DateTime.now().toIso8601String(),
-                          'screen_name': Pages.searchForCoin.name,
-                        });
+      builder: (context, coinState) {
+        return BlocConsumer<OrganisationDetailsCubit, OrganisationDetailsState>(
+          listener: (context, orgState) async {
+            log('orgState" $orgState');
 
-                    context.pushNamed(Pages.profileSelectionCoin.name);
-                  },
-                )
-              : null,
+            if (orgState is OrganisationDetailsSetState) {
+              log("Organisation is set: ${orgState.organisation.name}");
+              AnalyticsHelper.logEvent(
+                  eventName: AmplitudeEvent.nfcScanned,
+                  eventProperties: {
+                    'goal_name': orgState.organisation.name,
+                  });
+              coinCubit.stopAnimation();
+            }
+            if (orgState is OrganisationDetailsErrorState) {
+              coinCubit.error();
+              ScaffoldMessenger.of(context).showSnackBar(
+                SnackBar(
+                  content: const Text(
+                    "Error setting organisation. Please try again later.",
+                    textAlign: TextAlign.center,
+                  ),
+                  backgroundColor: Theme.of(context).errorColor,
+                ),
+              );
+            }
+          },
+          builder: (BuildContext context, orgState) {
+            log('builder: orgState: $orgState');
+            return Scaffold(
+              backgroundColor: const Color(0xFFEEEDE4),
+              body: coinState.status == CoinAnimationStatus.animating
+                  ? const SearchingForCoinPage()
+                  : coinState.status == CoinAnimationStatus.error
+                      ? const CoinErrorPage()
+                      : const CoinFoundPage(),
+              floatingActionButtonLocation:
+                  FloatingActionButtonLocation.centerFloat,
+              floatingActionButton:
+                  coinState.status == CoinAnimationStatus.stoped
+                      ? FloatingActoinButton(
+                          text: "Assign the coin",
+                          onPressed: () {
+                            AnalyticsHelper.logEvent(
+                                eventName: AmplitudeEvent.buttonPressed,
+                                eventProperties: {
+                                  'button_name': 'Assign the coin',
+                                  'formatted_date':
+                                      DateTime.now().toIso8601String(),
+                                  'screen_name': Pages.searchForCoin.name,
+                                });
+
+                            context.pushReplacementNamed(
```

Comment:
if we use pushReplacementNamed here, the back button on the profile selection screen does not work.
We need either remove back button from that screen, or change the navigation to just pushNamed (but in this case we need to handle it for the searchForCoin cubit/page)

Reply:
I don't think there is a reason from UX perspective to go back to the coin animation screen. Instead I think that back arrow should lead to the 'Homescreen'. 
The way we test the coinflow by long tap uses: pushNamedReplacement and it causes an error. I think if the coin-flow is activated from an nfc it would be an issue too in some cases.
So as a solution I added minor logic so if the BackButton cannot pop it redirects to splash which would either redirect to a log in, profile selection, or wallet screen. 
[https://github.com/givtnl/givt-app-kids/pull/127/files#diff-6c8ec6675c96f61181cf2b945065bfda76ea18d9b13d713430b99d33d34d52a0](https://github.com/givtnl/givt-app-kids/pull/127/files#diff-6c8ec6675c96f61181cf2b945065bfda76ea18d9b13d713430b99d33d34d52a0)